### PR TITLE
[docs] Getting Started with TVM: Auto Tuning with Python

### PR DESCRIPTION
**Pull request by hogepodge**
Follows up on the TVMC tutorial, shows how to accomplish the same tasks using the python api

### CODE REVIEW

**Comment by hogepodge** [comment]
@tqchen @leandron

**Comment by leandron** [comment]
cc @mbaret

File: tutorials/get_started/auto_tuning_with_python.py [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+#   By default this search is guided using an `XGBoost Grid` algorithm.
+#   Depending on your model complexity and amount of time avilable, you might
+#   want to choose a different algorithm.
+
+
+################################################################################
+# .. note:: Setting Tuning Parameters
+#
+#   In this example, in the interest of time, we set the number of trials and
+#   early stopping to 100. You will likely see more performacne improvements if
```
Comment (leandron) [inline tutorials/get_started/auto_tuning_with_python.py:321]:
```suggestion
#   early stopping to 100. You will likely see more performance improvements if
```

File: tutorials/get_started/auto_tuning_with_python.py [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+#   # [Task 22/24]  Current/Best:   13.33/ 207.66 GFLOPS | Progress: (1000/1000) | 761.74 s Done.
+#   # [Task 23/24]  Current/Best:   53.29/ 192.98 GFLOPS | Progress: (1000/1000) | 799.90 s Done.
+#   # [Task 24/24]  Current/Best:   25.03/ 146.14 GFLOPS | Progress: (1000/1000) | 1112.55 s Done.
+
+################################################################################
+# Compiling an Optimized Model with Tuning Data
+# ----------------------------------------------
+#
+# As an output of the tuning process above, we obtained the tuning records
+# stored in ``autotuner_records.json``. The compiler will use the results to
```
Comment (leandron) [inline tutorials/get_started/auto_tuning_with_python.py:375]:
```suggestion
# stored in ``resnet-50-v2-autotuning.json``. The compiler will use the results to
```

File: tutorials/get_started/auto_tuning_with_python.py [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+#
+
+import onnx
+from tvm.contrib.download import download_testdata
+from PIL import Image
+import numpy as np
+import tvm.relay as relay
+import tvm
+from tvm.contrib import graph_runtime
+import timeit
```
Comment (leandron) [inline tutorials/get_started/auto_tuning_with_python.py:65]: Maybe, for clarity, lazy import `timeit` it when you're discussing performance, similar to what it is done in the training section?

File: tutorials/get_started/auto_tuning_with_python.py [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+
+################################################################################
+# Downloading  and Loading  the ONNX Model
```
Comment (leandron) [inline tutorials/get_started/auto_tuning_with_python.py:68]: Are double spaces here intentional?

File: tutorials/get_started/auto_tuning_with_python.py [hunk 5]
```diff
@@ -0,0 +1,446 @@
...
+
+################################################################################
+# Downloading, Preprocessing, and Loading the Test Image
+# ------------------------------------------------------
+#
+# Each model is particular when it comes to expected tensor shapes, formats and
+# data types. For this reason, most models require some pre and
+# post-processing, to ensure the input is valid and to interpret the output.
+# TVMC has adopted NumPy's ``.npz`` format for both input and output data. This
+# is a well-supported NumPy format to serialize multiple arrays into a file
```
Comment (leandron) [inline tutorials/get_started/auto_tuning_with_python.py:104]: This is probably not relevant here.

File: tutorials/get_started/auto_tuning_with_python.py [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+
+################################################################################
+# Collect Basic Performance Data
+# ------------------------------
+# We want to collect some basic performance data associated with this
+# unoptimized model and compare it to a tuned model later.  To help account for
+# CPU noise, we run the computation in multiple batches in multiple
+# repetitions, then gather some basis statistics on the mean, median, and
+# standard deviation.
+
```
Comment (leandron) [inline tutorials/get_started/auto_tuning_with_python.py:199]: I suggest `import timeit` to appear here.

File: tutorials/get_started/auto_tuning_with_python.py [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+#   # class='n04040759 radiator' with probability=0.000261
+
+################################################################################
+# Comparing the Tuned and Untuned Models
+# --------------------------------------
+# We want to collect some basic performance data associated with this optimized
+# model to compare it to the unoptimized model. Depending on your underlying
+# hardware, number of iterations, and other factors, you should see a performance
+# improvement in comparing the optimized model to the unoptimized model.
+
```
Comment (leandron) [inline tutorials/get_started/auto_tuning_with_python.py:420]: Maybe for completeness, another `import timeit` here as well.

**Review by leandron — COMMENTED** [review]
Looks great, and give people a great perspective on how to use the Python API. Just a few minor comments.

**Comment by hogepodge** [comment]
@tqchen @leandron ready for final review

**Review by leandron — APPROVED** [review]
LGTM, with minor nit.

**Review by leandron — APPROVED** [review]
LGTM

**Comment by hogepodge** [comment]
Merge conflicts have turned this into a mess. I'm going to abandon and push a clean patch.

**Pull request closed without merging**